#!/usr/bin/env python

PACKAGE = "aruco_cube_ros"
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t

gen = ParameterGenerator()

gen.add("stream_image", bool_t, 0, "Detect aruco from image stream.", True)
gen.add("display_opencv_window", bool_t, 0, "Display opencv window.", True)
gen.add("broadcast_transform", bool_t, 0, "Broadcast transform of aruco cube.", False)
gen.add("publish_pose", bool_t, 0, "Publish pose of aruco cube.", True)
gen.add("window_size", int_t, 0, "Window size for pose filtering.", 0, 0, 200)
gen.add("do_corner_refinement", bool_t, 0, "Do corner refinement.", True)
gen.add("corner_refinement_win_size", int_t, 0, "Window size ofsubpixel refinement process.", 2, 0, 20)
gen.add("corner_refinement_max_iterations", int_t, 0, "Maximum number of interations of subpixel refinement process.", 50, 0, 200)
gen.add("corner_refinement_min_accuracy", double_t, 0, "Minimum erorr value of subpixel refinement process.", 0.9, 0, 1)
gen.add("max_erroneous_bits_in_border_rate", double_t, 0, "Max erroneous bits in boarder.", 0.35, 0, 1)
gen.add("error_correction_rate", double_t, 0, "Error correction rate.", 0.6, 0, 1)

exit(gen.generate(PACKAGE, "aruco_cube", "ArucoCube"))
